src/Common/tests/System/AssertExtensions.cs

```diff
@@ -14,5 +15,17 @@ public static void Throws<T>(Action action, string message)
         {
             Assert.Equal(Assert.Throws<T>(action).Message, message);
         }
+
+        public static void Throws<T>(string netCoreParamName, string netFxParamName, Action action)
+            where T : ArgumentException
+        {
+            T exception = Assert.Throws<T>(action);
+
+            string expectedParamName =
+                RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework") ?
+                netFxParamName : netCoreParamName;
```

Comment:
@weshaggard, do we still run tests against previous versions of .NET Core?  If yes, this will likely hit failures in a bunch of the changed cases.  If no, it's fine.